### PR TITLE
AfterEffects: support for workfile template builder

### PR DESCRIPTION
**Pull request by kalisp**
## Changelog Description
This PR add functionality of templated workfile builder. It allows someone to prepare AE workfile with placeholders as for automatically loading particular representation of particular subset of particular asset from context where workfile is opened.

Selection from multiple prepared workfiles is provided with usage of templates, specific type of tasks could use particular workfile template etc.

Artists then can build workfile from template when opening new workfile.

## Testing notes:
!!! It requires to reinstall extension !!!
Watch this video 

https://github.com/ynput/OpenPype/assets/4457962/f5e1bce9-f57d-4fe8-beb7-63d57e7a2096

### CODE REVIEW

**Comment by jakubjezek001** [comment]
Task linked: [OP-5661 AE: template builder](https://app.clickup.com/t/6658547/OP-5661)

**Review by iLLiCiTiT — APPROVED** [review]
Code LGTM. Didn't test it.

File: openpype/hosts/aftereffects/api/extension/js/main.js [hunk 1]
```diff
@@ -280,14 +300,24 @@ function main(websocket_url){
...
+                                                      data.item_id + "," +
                                                   " null )")
            .then(function(result){
                log.warn("addItemAsLayerToComp: " + result);
                return result;
            });
    });
 
+   RPC.addRoute('AfterEffects.add_item_instead_placeholder', function (data) {
+    log.warn('Server called client route "add_item_instead_placeholder":', data);
```
Comment (Minkiu) [inline openpype/hosts/aftereffects/api/extension/js/main.js:312]: Different identation (not that it matter in js but hey)

File: openpype/hosts/aftereffects/api/extension/jsx/hostscript.jsx [hunk 2]
```diff
@@ -139,6 +165,24 @@ function getItems(comps, folders, footages){
...
 }
 
+function selectItems(items){
+    /**
+     * Select all items from `items`, deselect other.
+     *
+     * Args:
+     *      items (list)
+     */
+    for (i = 1; i <= app.project.items.length; ++i){
```
Comment (Minkiu) [inline openpype/hosts/aftereffects/api/extension/jsx/hostscript.jsx:175]: Does jsx/after effects allow to use `app.project.items.forEach()` ?

**Review by Minkiu — APPROVED** [review]
Haven't tested it, but code looks good.

**Review by 64qam — APPROVED** [review]

**Pull request merged**